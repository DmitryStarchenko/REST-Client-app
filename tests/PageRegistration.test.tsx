import { render, screen } from '@testing-library/react';
import React from 'react';
import { describe, it, expect, vi } from 'vitest';

const mockSignUp = vi.hoisted(() => vi.fn());

vi.mock('next-intl', () => ({
  useTranslations: vi.fn(() => (key: string) => key),
}));

vi.mock('@/i18n/navigation', () => ({
  Link: vi.fn(({ children, href, className }) =>
    React.createElement(
      'a',
      {
        href,
        className,
      },
      children,
    ),
  ),
  useRouter: vi.fn(() => ({
    push: vi.fn(),
  })),
  usePathname: vi.fn(() => '/'),
}));

vi.mock('../src/lib/supabase/client', () => ({
  default: {
    auth: {
      signUp: mockSignUp,
    },
  },
}));

vi.mock('@/components/auth/RegistrationForm/RegistrationForm', () => ({
  default: vi.fn(() => <div data-testid="registration-form">Registration Form</div>),
}));

import Registration from '../src/app/[locale]/(auth)/registration/page';

describe('Registration Page', () => {
  it('should render without crashing', () => {
    render(<Registration />);
    const registrationForm = screen.queryByTestId('registration-form');
    if (registrationForm) {
      expect(screen.getByTestId('registration-form')).toBeInTheDocument();
    }
  });

  it('should render RegistrationForm component', () => {
    render(<Registration />);
    const registrationForm = screen.queryByTestId('registration-form');
    if (registrationForm) {
      const registrationForm = screen.getByTestId('registration-form');
      expect(registrationForm).toBeInTheDocument();
      expect(registrationForm).toHaveTextContent('Registration Form');
    }
  });

  it('should match snapshot', () => {
    const { container } = render(<Registration />);
    expect(container).toMatchSnapshot();
  });
});
