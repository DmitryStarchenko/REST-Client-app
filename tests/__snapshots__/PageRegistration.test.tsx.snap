// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Registration Page > should match snapshot 1`] = `
<div>
  <form
    class="_form_e19e11"
  >
    <h1>
      Registration
    </h1>
    <div
      class="MuiFormControl-root MuiTextField-root css-1xp5r68-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-2hy7sk-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="email"
        id="email-label"
      >
        email
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-5h82ro-MuiInputBase-root-MuiInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input css-1yrc8ca-MuiInputBase-input-MuiInput-input"
          id="email"
          name="email"
          type="text"
        />
      </div>
    </div>
    <div
      class="MuiFormControl-root css-1iw3t7y-MuiFormControl-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-2hy7sk-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="standard-adornment-password"
      >
        password
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-6h9cc6-MuiInputBase-root-MuiInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd css-1yrc8ca-MuiInputBase-input-MuiInput-input"
          id="password"
          name="password"
          type="password"
        />
        <div
          class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-elo8k2-MuiInputAdornment-root"
        >
          <button
            aria-label="display the password"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="VisibilityIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5M12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5m0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3"
              />
            </svg>
          </button>
        </div>
      </div>
      <p
        class="MuiFormHelperText-root MuiFormHelperText-sizeMedium css-18y2hez-MuiFormHelperText-root"
      />
    </div>
    <div
      class="MuiFormControl-root css-1iw3t7y-MuiFormControl-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-2hy7sk-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="standard-adornment-password"
      >
        confirm password
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-6h9cc6-MuiInputBase-root-MuiInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd css-1yrc8ca-MuiInputBase-input-MuiInput-input"
          id="confirmPassword"
          name="confirmPassword"
          type="password"
        />
        <div
          class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-elo8k2-MuiInputAdornment-root"
        >
          <button
            aria-label="display the password"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="VisibilityIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5M12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5m0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3"
              />
            </svg>
          </button>
        </div>
      </div>
      <p
        class="MuiFormHelperText-root MuiFormHelperText-sizeMedium css-18y2hez-MuiFormHelperText-root"
      />
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1uent87-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="submit"
    >
      <span
        class="MuiButton-loadingWrapper"
        style="display: contents;"
      />
      Submit
    </button>
  </form>
</div>
`;
